code/src/components/todolist.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import { ToDoInput } from './todoinput';
+import { ToShopInput } from './toshopinput';
+import { ToDoItem } from './todoitem';
+import { Summary } from './todosummary';
+import { useSelector, useDispatch } from 'react-redux';
+import { todos } from '../reducers/todos.js';
+import { CompletedTasks, CompletedShopping } from './completedtodolist';
+import '../index.css';
```

Comment:
Try structuring your imports so that the external imports come first, and then your local ones (move react-redux below react)